packages/core/src/new-api/internal/journal/file-journal.ts

```diff
@@ -0,0 +1,42 @@
+import fs from "fs";
+import ndjson from "ndjson";
+
+import { Journal, JournalableMessage } from "../../types/journal";
+
+import { deserializeReplacer } from "./utils/deserialize-replacer";
+import { serializeReplacer } from "./utils/serialize-replacer";
+
+/**
+ * A file-based journal.
+ *
+ * @beta
+ */
+export class FileJournal implements Journal {
+  constructor(private _filePath: string) {}
+
+  public async record(message: JournalableMessage): Promise<void> {
+    return fs.promises.appendFile(
```

Comment:
I think we should do something like this to write:

```ts
function appendJsonLine(path: string, value: unknown) {
  const fd = openSync(
    path,
    constants.O_CREAT | // Create it if it doesn't exist
      constants.O_WRONLY | // Write only
      constants.O_APPEND | // Append
      constants.O_DSYNC | // Synchronous I/O waiting for writes of content and metadata
      constants.O_DIRECT // Minimize caching
  );

  writeFileSync(fd, `\n${JSON.stringify(value)}`, "utf-8");
  closeSync(fd);
}
```

Reply:
Note that it would always leave an empty first line, which i guess should be ok

Reply:
I have switched to this implementation and made the `record` function sync.